src/Symfony/Component/Security/Core/Role/SwitchUserRole.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Role;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+/**
+ * Allows migrating session payloads from v4 to v5.
+ *
+ * @internal to be removed in Symfony 6.0
+ */
+class SwitchUserRole extends Role
+{
+    private $deprecationTriggered;
```

Comment:
Is this property needed?

Reply:
It exists on v4. No declaring it would make unserialization add it as a dynamic property. Better not.